### PR TITLE
httphandler: Initial on-the-fly image resize HTTP endpoint

### PR DESCRIPTION
**Pull request by xor-gate**
This is a initial attempt to have ad-hoc resizing without disk-cache.

### CODE REVIEW

File: httphandler.go [hunk 1]
```diff
@@ -36,6 +39,34 @@ func HttpHandlerInstallEndpoints(db *storm.DB) {
...
+		}
+		defer file.Close()
+
+		img, err := jpeg.Decode(file)
+		if err != nil {
+			return
+		}
+
+		// Resize to ratio with a max width of 1000 pixels
+		m := resize.Resize(1000, 0, img, resize.Lanczos3)
```
Comment (rikvdh) [inline httphandler.go]: Perhaps add 1000 as an optional second parameter to the URL

File: httphandler.go [hunk 2]
```diff
@@ -36,6 +39,34 @@ func HttpHandlerInstallEndpoints(db *storm.DB) {
...
+		jpeg.Encode(w, m, nil)
```
Comment (rikvdh) [inline httphandler.go]: Don't you want the content-disposition headers before emitting the JPEG to the user. :-)

Comment (xor-gate) [inline httphandler.go]: Thats a nice catch, users can download resized pictures but then they will be saved as the name of the picture id.

File: httphandler.go (hunk 1 above)
Comment (xor-gate) [inline httphandler.go]: That would be nice 👍

Comment (rikvdh) [inline httphandler.go]: Type faster!

**Pull request merged**